### PR TITLE
Shiver me timbers (fixed gun recoil showing for everyone)

### PR DESCRIPTION
**Pull request by Bod9001**
### Purpose
ayy we finally killed the kraken that's been shaking the ship, Crewe be now not sick as a dog!

### Changelog:

CL: [Fix] killed the kraken , (fixed gun recoil showing for everyone)

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/Weapons/Gun.cs [hunk 1]
```diff
@@ -729,15 +729,29 @@ private bool TryReload(GameObject ammo)
...
-			RPCShowMuzzleFlashAndRecoil(shooter.GetComponent<NetworkIdentity>(), finalDirection);
+			var Identity = shooter.GetComponent<NetworkIdentity>();
+			if (Identity.OrNull()?.connectionToClient != null)
+			{
+				if (isServer && shooter == PlayerManager.LocalPlayer)
+				{
+					Camera2DFollow.followControl.Recoil(-finalDirection, CameraRecoilConfig);
+				}
 
+				RPCShowMuzzleFlash(shooter.GetComponent<NetworkIdentity>());
```
Comment (Alainx277) [inline UnityProject/Assets/Scripts/Items/Weapons/Gun.cs]:
```suggestion
				RPCShowMuzzleFlash(Identity);
```

File: UnityProject/Assets/Scripts/Items/Weapons/Gun.cs [hunk 2]
```diff
@@ -729,15 +729,29 @@ private bool TryReload(GameObject ammo)
 				SoundManager.PlayNetworkedAtPos(FiringSoundA, shooter.transform.position);
 			}
 
-			RPCShowMuzzleFlashAndRecoil(shooter.GetComponent<NetworkIdentity>(), finalDirection);
+			var Identity = shooter.GetComponent<NetworkIdentity>();
```
Comment (Alainx277) [inline UnityProject/Assets/Scripts/Items/Weapons/Gun.cs]: `Identity` should probably be lowercase.

File: UnityProject/Assets/Scripts/Items/Weapons/Gun.cs (hunk 1 above)
Comment (Alainx277) [inline UnityProject/Assets/Scripts/Items/Weapons/Gun.cs]: Muzzle flash should be shown even if `connectionToClient` is `null`, as it is visible to everyone.

**Review by Alainx277 — CHANGES_REQUESTED** [review]

**Review by Alainx277 — APPROVED** [review]

File: UnityProject/Assets/Scripts/Items/Weapons/Gun.cs [hunk 3]
```diff
@@ -729,15 +729,28 @@ private bool TryReload(GameObject ammo)
...
-			RPCShowMuzzleFlashAndRecoil(shooter.GetComponent<NetworkIdentity>(), finalDirection);
+			var identity = shooter.GetComponent<NetworkIdentity>();
+			RPCShowMuzzleFlash(identity);
+			if (identity.OrNull()?.connectionToClient != null)
+			{
+				if (isServer && shooter == PlayerManager.LocalPlayer)
+				{
+					Camera2DFollow.followControl.Recoil(-finalDirection, CameraRecoilConfig);
+				}
+				RPCShowRecoil(identity.connectionToClient , finalDirection);
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Items/Weapons/Gun.cs:740]: The target RPC doesn't work for the host? That's a little surprising.

**Review by NoooneyDude — APPROVED** [review]

**Pull request merged**